app/services/sales_tax_service.rb

```
@@ -34,6 +35,13 @@ def record_tax_collected
     raise Errors::ProcessingError.new(:tax_recording_failure, message: e.message)
   end
 
+  def refund_transaction(refund_date)
+    @transaction = get_transaction(transaction_id) if @transaction.blank?
+    @refund = post_refund(refund_date) if @transaction.present?
```

should we `raise` a `ProcessingError` if `@transaction` was not present? that way once we move this job to sidekiq it will get automatic retries. Thinking of the case where Taxjar was down or something

---

If TaxJar is down then we'll get a different error than `Taxjar::Error::NotFound`. It's not an error state if `@transaction` is missing since a given refund may or may not include a transaction. We could infer if a refund should include a transaction from `line_item.should_remit_sales_tax` or from `artsy_should_remit_tax?` but I think it's safer to just always try so that we don't inadvertently leave something out in case the logic changes later on.